connect/http-client/src/main/java/org/camunda/connect/httpclient/impl/AbstractHttpConnector.java

```diff
@@ -69,15 +70,33 @@ public void setHttpClient(CloseableHttpClient httpClient) {
   @Override
   public R execute(Q request) {
     HttpRequestBase httpRequest = createHttpRequest(request);
-
     HttpRequestInvocation invocation = new HttpRequestInvocation(httpRequest, request, requestInterceptors, httpClient);
-
     try {
-      return createResponse((CloseableHttpResponse) invocation.proceed());
-    } catch (Exception e) {
-      throw LOG.unableToExecuteRequest(e);
-    }
+	  
+      R invocationResult = createResponse((CloseableHttpResponse) invocation.proceed());
```

Comment:
🔧 Let's restructure this because this way the http error exception is caught and rethrown as `unableToExecuteRequest`: Move the new logic out from the try catch so it's not caught:

```java
R invocationResult;
try {
  invocationResult = createResponse((CloseableHttpResponse) invocation.proceed());
} catch (Exception e) {
  throw LOG.unableToExecuteRequest(e);
}

handleErrorResponse(request, invocationResult);
return invocationResult;
```